### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #ljust
- #rjust
- #center
- #partition
- #rpartition

### CODE REVIEW

File: doc/string/center.rdoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  'hello'.center(10)       # => "  hello   "
+  '  hello'.center(10)     # => "   hello  "
+  'hello'.center(10, 'ab') # => "abhelloaba"
+  'тест'.center(10)        # => "   тест   "
+  'こんにちは'.center(10)    # => "  こんにちは   "
+
+If +size+ is not greater than the size of +self+, returns a copy of +self+:
+
+  'hello'.center(5)  # => "hello"
+  'hello'.center(1)  # => "hello"
```
Comment (peterzhu2118) [inline doc/string/center.rdoc:16]: For each of these, should we add a "Related" section to it to suggest the other options?

```suggestion
  'hello'.center(1)  # => "hello"

Related: String#ljust, String#rjust
```

Comment (BurdetteLamar) [reply doc/string/center.rdoc:16]: Done.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Comment by Maumagnaguagno** [comment]
You are also modifying ``partition`` and ``rpartition`` docs, they are not listed in the description.
Related to #5587.

File: doc/string/rpartition.rdoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  'hello'.rpartition('l')      # => ["hel", "l", "o"]
+  'hello'.rpartition('ll')     # => ["he", "ll", "o"]
+  'hello'.rpartition('h')      # => ["", "h", "ello"]
+  'hello'.rpartition('o')      # => ["hell", "o", ""]
+  'hello'.rpartition('')       # => ["hello", "", ""]
+  'тест'.rpartition('т')       # => ["тес", "т", ""]
+  'こんにちは'.rpartition('に')  # => ["こん", "に", "ちは"]
+
+If the pattern is not matched, returns a copy of +self+ and two empty strings:
```
Comment (Maumagnaguagno) [inline doc/string/rpartition.rdoc:18]: Match the order in which empty strings appear in the resulting array:
``If the pattern is not matched, returns two empty strings and a copy of +self+:``

File: doc/string/partition.rdoc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+If the pattern is matched, returns pre-match, first-match, post-match:
+
+  'hello'.partition('l')      # => ["he", "l", "lo"]
+  'hello'.partition('ll')     # => ["he", "ll", "o"]
+  'hello'.partition('h')      # => ["", "h", "ello"]
+  'hello'.partition('o')      # => ["hell", "o", ""]
+  'hello'.partition('')       # => ["", "", "hello"]
+  'тест'.partition('т')       # => ["", "т", "ест"]
+  'こんにちは'.partition('に')  # => ["こん", "に", "ちは"]
```
Comment (peterzhu2118) [inline doc/string/partition.rdoc:16]: I think we should have a regex example, e.g.

```ruby
'hello'.partition(/l+/) #=> ["he", "ll", "o"]
```

File: doc/string/partition.rdoc [hunk 4]
```diff
@@ -0,0 +1,22 @@
+Returns a 3-element array of substrings of +self+.
+
+Matches a pattern against +self+; the pattern is:
```
Comment (peterzhu2118) [inline doc/string/partition.rdoc:3]: I think the wording of the description should be different for `String#partition` and `String#rpartition` to highlight the difference.

```suggestion
Matches a pattern against +self+ from the beginning, scanning forwards. The pattern is:
```

File: doc/string/rpartition.rdoc [hunk 5]
```diff
@@ -0,0 +1,22 @@
+Returns a 3-element array of substrings of +self+.
+
+Matches a pattern against +self+; the pattern is:
```
Comment (peterzhu2118) [inline doc/string/rpartition.rdoc:3]:
```suggestion
Matches a pattern against +self+ from the end, scanning backwards. The pattern is:
```

File: doc/string/rpartition.rdoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+If the pattern is matched, returns pre-match, last-match, post-match:
+
+  'hello'.rpartition('l')      # => ["hel", "l", "o"]
+  'hello'.rpartition('ll')     # => ["he", "ll", "o"]
+  'hello'.rpartition('h')      # => ["", "h", "ello"]
+  'hello'.rpartition('o')      # => ["hell", "o", ""]
+  'hello'.rpartition('')       # => ["hello", "", ""]
+  'тест'.rpartition('т')       # => ["тес", "т", ""]
+  'こんにちは'.rpartition('に')  # => ["こん", "に", "ちは"]
+
```
Comment (peterzhu2118) [inline doc/string/rpartition.rdoc:17]: I think it's valuable to have the following comment from the original description:

>  The match from the end means starting at the possible last position, not the last of longest matches.

And the following example:

```ruby
'hello'.rpartition(/l+/) #=> ["hel", "l", "o"]
```

File: doc/string/rpartition.rdoc (hunk 2 above)
Comment (BurdetteLamar) [reply doc/string/rpartition.rdoc]: Fixed.

File: doc/string/partition.rdoc (hunk 4 above)
Comment (BurdetteLamar) [reply doc/string/partition.rdoc]: Fixed.

File: doc/string/rpartition.rdoc (hunk 5 above)
Comment (BurdetteLamar) [reply doc/string/rpartition.rdoc]: Fixed.

**Comment by BurdetteLamar** [comment]
Meant also to say in original description: I've left out the "lookbehind" in rpartition as being overly esoteric.  If it should be left in, should there be something corresponding in partition?

**Comment by BurdetteLamar** [comment]
@Maumagnaguagno, the re-review button for your review was dimmed;  don't know why..

**Comment by peterzhu2118** [comment]
I'm fine with both keeping and removing the lookbehind part. On one hand, I think showing more corner cases is a good thing, but on the other hand I do think it's a really corner case.

**Pull request merged**